amf-shapes/shared/src/main/scala/amf/shapes/internal/spec/common/parser/TypeDeclarationParser.scala

```diff
@@ -4,18 +4,50 @@ import amf.aml.internal.parse.common.{DeclarationKey, DeclarationKeyCollector}
 import amf.core.client.scala.model.domain.AmfScalar
 import amf.core.internal.annotations.DeclaredElement
 import amf.core.internal.metamodel.domain.ShapeModel
-import amf.core.internal.parser.YMapOps
 import amf.core.internal.parser.domain.Annotations
+import amf.core.internal.parser.{YMapOps, _}
 import amf.shapes.client.scala.model.domain.{AnyShape, NodeShape}
 import amf.shapes.internal.spec.oas.parser.OasTypeParser
-import amf.shapes.internal.validation.definitions.ShapeParserSideValidations.UnableToParseShape
+import amf.shapes.internal.validation.definitions.ShapeParserSideValidations.{MultipleDefinitionKey, UnableToParseShape}
 import org.yaml.model.{YMap, YScalar}
-import amf.core.internal.parser._
 
 object TypeDeclarationParser {
 
   def parseTypeDeclarations(map: YMap, definitionsKey: String, declarationKeysHolder: Option[DeclarationKeyCollector])(
       implicit ctx: ShapeParserContext
+  ): List[AnyShape] = parseTypeDeclarations(map, Seq(definitionsKey), declarationKeysHolder)
+
+  def parseTypeDeclarations(
+      map: YMap,
+      definitionsKeys: Seq[String],
+      declarationKeysHolder: Option[DeclarationKeyCollector]
+  )(implicit ctx: ShapeParserContext): List[AnyShape] = {
+    validateMultipleDeclarationKeys(map, definitionsKeys)
+    definitionsKeys.flatMap(defKey => parseDeclarationMap(map, defKey, declarationKeysHolder)).toList
+  }
+
+  private def validateMultipleDeclarationKeys(map: YMap, definitionsKeys: Seq[String])(implicit
+      ctx: ShapeParserContext
+  ): Unit = {
+    val foundDefKeys = definitionsKeys.flatMap(map.key(_))
+    if (foundDefKeys.size > 1) { // If there is more than 1 definition key present in the map
+      // TODO add test in the new validation suite
+      ctx.eh.warning(
```

Comment:
If we are going to parse only one key when both are presents, maybe should be a violation instead of a warning?
I think when some information is lost (not parsed) we should throw violations, because some information is not evaluated (and those ignored ast may be invalid at amf)

Reply:
We are parsing all the keys (or at least I suppose I do that)

Reply:
`definitionsKeys.flatMap(defKey => parseDeclarationMap(map, defKey, declarationKeysHolder)).toList `
All the declarations in all the keys should be parsing